polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1245,6 +1245,23 @@ void CodeGenTypes::constructAttributeList(
   AttrList.addAttrs(FuncAttrsBuilder, RetAttrsBuilder, ArgAttrs);
 }
 
+mlir::Type CodeGenTypes::getMLIRTypeForMem(clang::QualType QT,
+                                           bool *ImplicitRef, bool AllowMerge) {
+  assert(!QT->isConstantMatrixType() && "Unsupported type");
+
+  const auto R = getMLIRType(QT, ImplicitRef, AllowMerge);
+
+  // TODO: Check for the boolean vector case.
+
+  // If this is a bool type map this integer to the target-specified size.
+  if (R.isInteger(1))
```

Comment:
Looking at the implementation in `CodeGenTypes.cpp` (used to generate LLVM IR), should this condition be:

```
if (!QT->isBitIntType() && R.isInteger(1))

Reply:
Well, that's part of the original code. I just avoided including it because I wasn't handling `_BitInt` here (also, you cannot have 1 bit `_BitInt`, so I'm pretty sure that condition's always false). I'll add it for symmetry and just in case.